src/gsi/gsimod.F90

```diff
@@ -1757,6 +1765,9 @@ subroutine gsimain_initialize
      end if
   end if
 
+! Ensure when running 2DVAR (RTMA) or 3DRTMA, writing out obs provider info
+  if (twodvar_regional .or. l_rtma3d) l_obsprvdiag = .true.
+
```

Comment:
It is better NOT to force (or hardcoded) this inside the code as it hide information from general users.
For 2D/3D RTMA, setting l_obsprvdiag = .true. in the namelist is enough and provides a flexibility/backward compatibility to NOT to output subprovider information.

Reply:
I will remove this line.